### DISCUSSION TITLE
'Your first 3D game'  tutorial

### DISCUSSION THREAD

**Discussion by Tauhong** [godotengine/godot-proposals#7615] [3D]
Hi there, I'm new to Godot.

First off, thanks for putting up some really great documentation.

Just wondering if you could improve this bit in the player movement section

if direction != Vector3.ZERO:
        direction = direction.normalized()
        $Pivot.look_at(position + direction, Vector3.UP)

Would it be possible to add a couple of lines to explain why $Pivot.look_at(position + direction, Vector3.UP) is necessary? 

Thank you.

**Comment by AThousandShips** [comment]
It says below:
> In this case, we also get the Pivot node and call its look_at() method. This method takes a position in space to look at in global coordinates and the up direction. In this case, we can use the Vector3.UP constant.
>
> In 3D, the property that contains a node's position is position. By adding the direction to it, we get a position to look at that's one meter away from the Player.

**Reply by Tauhong** [reply]
I did read that, yes, thank you. 

However, what I meant was - why are we 'look(ing) at global coordinates and up direction' at this particular spot in the code? What's the purpose of doing that?

**Reply by AThousandShips** [reply]
To get the player to face where we are moving, hence why we add the direction we are moving in to our position, if you run the code you'll see the behaviour clearly

I don't think the exact reason for doing this needs to be spelled out

The code is wrong however and uses `position` rather than `global_position`, it just happens to be equivalent because of how `Player` is positioned.

**Reply by AThousandShips** [reply]
Making some changes to this code and will clarify with the comment used in the examples:
> In the lines below, we turn the character, ...

With some modifications

**Comment by Tauhong** [comment]
Thank you for considering and acting on the request.
